umake/frameworks/swift.py

```diff
@@ -0,0 +1,120 @@
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+
+"""Swift module"""
+
+from contextlib import suppress
+from gettext import gettext as _
+import logging
+import os
+import re
+import lsb_release
+import gnupg
+
+import umake.frameworks.baseinstaller
+from umake.interactions import DisplayMessage
+from umake.tools import get_current_arch, add_env_to_user, ChecksumType, MainLoop
+from umake.network.download_center import DownloadCenter, DownloadItem
+from umake.ui import UI
+
+logger = logging.getLogger(__name__)
+
+
+class SwiftCategory(umake.frameworks.BaseCategory):
+
+    def __init__(self):
+        super().__init__(name="Swift", description=_("Swift language"),
+                         logo_path=None)
+
+
+class SwiftLang(umake.frameworks.baseinstaller.BaseInstaller):
+
+    def __init__(self, category):
+        super().__init__(name="Swift Lang", description=_("Swift compiler (default)"), is_category_default=True,
+                         packages_requirements=["clang", "python3-gnupg"],
+                         category=category, only_on_archs=['amd64'],
+                         download_page="https://swift.org/download/",
+                         dir_to_decompress_in_tarball="swift*",
+                         required_files_path=[os.path.join("usr", "bin", "swift")])
+        self.release = lsb_release.get_distro_information()['RELEASE']
+
+    def parse_download_link(self, line, in_download):
+        """Parse Swift download link, expect to find a .sig file"""
+        url_found = False
+        if self.release in line and '.tar.gz.sig' in line:
+            in_download = True
+        else:
+            in_download = False
+        if in_download:
+            p = re.search(r'href="(.*)" title="PGP Signature"', line)
+            with suppress(AttributeError):
+                self.sig_url = "https://swift.org" + p.group(1)
+                logger.debug(self.sig_url)
+                url_found = True
+                DownloadCenter(urls=[DownloadItem(self.sig_url, None)],
+                               on_done=self.check_gpg_and_start_download, download=False)
+        return (url_found, in_download)
+
+    @MainLoop.in_mainloop_thread
+    def get_metadata_and_check_license(self, result):
+        """Download files to download + license and check it"""
+        logger.debug("Parse download metadata")
+
+        error_msg = result[self.download_page].error
+        if error_msg:
+            logger.error("An error occurred while downloading {}: {}".format(self.download_page, error_msg))
+            UI.return_main_screen(status_code=1)
+
+        in_download = False
+        url_found = False
+        for line in result[self.download_page].buffer:
+            line_content = line.decode()
+            if not url_found:
+                (_url_found, in_download) = self.parse_download_link(line_content, in_download)
+            if not url_found:
```

Comment:
You have twice `if not url_found:` following each other without assignement in between, so, that can be replaced by with:
```
                (url_found, in_download) = self.parse_download_link(line_content, in_download)
```
(or maybe you can have the logic with "if this version is higher and lower than expected release here…"